### PR TITLE
added purchase url parser

### LINKED ISSUE #2345: Establish QR structure to openup account boosting purchase screen.
Design modal for Boosting account screen

### PR DESCRIPTION
**Pull request by aliseyalvi**
### What does this PR?
This PR adds support for parsing purchase urls and navigate user to account boost screen when scanning a valid purchase url

### Issue number
fixes #2345 

### Screenshots/Video

Uploading Screenrecorder-2022-06-09-20-42-49-64.mp4…

### CODE REVIEW

File: src/utils/deepLinkParser.ts [hunk 1]
```diff
@@ -80,6 +81,18 @@ export const deepLinkParser = async (url, currentAccount) => {
...
       keey = `${mode}/${referredUser || ''}`;
     }
+
+    //if url is for purchase
+    const { type, username} = parsePurchaseUrl(url) || {};
+    console.log('type, username : ', type, username);
+    if(type && type === 'boost'){
+      routeName = ROUTES.SCREENS.ACCOUNT_BOOST;
+      params = {
+        username,
```
Comment (noumantahir) [inline src/utils/deepLinkParser.ts:91]: accountBoostScreen is not reading navigation parameters, so this username is being discarded... read it in accountBoostScreen and pass it to InAppPurchaseContainer. The prop should be optional but if preset should override the currentAccount.name already being used in InAppPurchases.

references
https://github.com/ecency/ecency-mobile/blob/d1fa36330399c05048cd546723fefc719d690f10/src/containers/inAppPurchaseContainer.ts#L84

https://github.com/ecency/ecency-mobile/blob/d1fa36330399c05048cd546723fefc719d690f10/src/screens/accountBoost/screen/accountBoostScreen.js#L27

**Review by noumantahir — CHANGES_REQUESTED** [review]

**Comment by aliseyalvi** [comment]
@noumantahir  updated.

- added username param in purchase screen
- added user avatar in purchase screen

image

**Comment by aliseyalvi** [comment]
updated: 
image

**Comment by noumantahir** [comment]
Fine Tuned UI for account boost screen
Screenshot 2022-06-10 at 5 35 59 PM

**Comment by noumantahir** [comment]
@feruzm you may test and merge this.... QR code for testing....
qrcode (1)
https://ecency.com/purchase?type=boost&username=demo.com

**Review by feruzm — APPROVED** [review]
lgtm

**Pull request merged**